src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioActiveDocumentTracker.cs

```diff
@@ -147,10 +147,17 @@ public void TrackNewActiveWindowFrame(IVsWindowFrame frame)
 
             _activeFrame = frame;
 
-            if (!_visibleFrames.Any(f => f.Frame == frame))
+            var existingFrame = _visibleFrames.FirstOrDefault(f => f.Frame == frame);
+            if (existingFrame == null)
             {
                 _visibleFrames = _visibleFrames.Add(new FrameListener(this, frame));
             }
+            else if (existingFrame.TextBuffer == null)
+            {
+                // If no text buffer is associated with existing frame, remove the existing frame and add the new one.
+                _visibleFrames = _visibleFrames.Remove(existingFrame);
```

Comment:
Do we need to disconnect the existing FrameListener, or we might end up creating multiple subscriptions?

Reply:
I guess we disconnect if the frame is disconnected or changed, so is the existing one perhaps already disconnected? This might be fine then (or if it's a leak, it'll clean up pretty quickly), but a comment to that effect would be good if that's what we're relying on.

Reply:
It is the latter - Disconnect gets called for all the frames when the tab is finally closed or de-activated. I have added a comment.